hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/BucketEndpoint.java

```diff
@@ -145,6 +145,10 @@ public Response get(
       } else {
         throw ex;
       }
+    } catch (OS3Exception ex) {
+      getMetrics().incGetBucketFailure();
+      LOG.debug("Bucket Does not Exist " + bucketName, ex);
+      throw newError(S3ErrorTable.NO_SUCH_BUCKET, bucketName, ex);
```

Comment:
We should just rethrow `ex`, because:

 * `OS3Exception` already has an error code and proper message.  No need to wrap in another exception.
 * Cause may be different, not necessarily `NO_SUCH_BUCKET`.

On second thought we also don't need the debug message, because the exception is logged at debug level in `newError`.

https://github.com/apache/ozone/blob/e81ab0a3f76b09072c23e0151c5e28673623ed6c/hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/exception/S3ErrorTable.java#L141-L145

```suggestion
      throw ex;
```

Reply:
+1 we should just increment the counter and retrhwo

Reply:
Sure will do !!